plugins/inputs/redfish/redfish.go

```diff
@@ -0,0 +1,312 @@
+package redfish
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io/ioutil"
+	"math"
+	"net/http"
+)
+
+type Cpu struct {
+	Name           string    `json:"Name"`
+	ReadingCelsius int64     `json:"ReadingCelsius"`
+	Status         CpuStatus `json:"Status"`
+}
+type Payload struct {
+	Temperatures  []*Cpu   `json:",omitempty"`
+	Fans          []*speed `json:",omitempty"`
+	PowerSupplies []*Watt  `json:",omitempty"`
+	Hostname      string   `json:",omitempty"`
+	Voltages      []*volt  `json:",omitempty"`
+	Location      *Address `json:",omitempty"`
+}
+type CpuStatus struct {
+	State  string
+	Health string
+}
+type speed struct {
```

Comment:
struct casing isn't consistent. Use Capital first letter for public structs, lowercase for private.